src/adaptation/icap/Xaction.cc

```diff
@@ -134,6 +134,22 @@ void Adaptation::Icap::Xaction::start()
     Adaptation::Initiate::start();
 }
 
+// TODO: Make reusable by moving this (and the printing operator from
+// ip/Address.h that this code is calling) into ip/print.h or similar.
```

Comment:
I marked this PR as a Draft because I was not sure whether this TODO should be addressed in this unofficial PR, in the corresponding official PR, or later, in a followup official PR. @yadij, what is your preference?

Reply:
This looks like a case for adding ip/gadgets.cc

If this builds without adding problems I am good with it here though.

Reply:
It does build without adding problems (for some definition of "adding problems") AFAICT. It just does not make this code reusable. I can make it reusable (i.e. add `ip/gadgets`) in this PR or in a dedicated PR. @yadij, do you have a preference between those two options?